pkg/monitortests/node/kubeletlogcollector/node.go

```diff
@@ -712,3 +722,66 @@ func getNodeLog(ctx context.Context, client kubernetes.Interface, nodeName, syst
 
 	return ioutil.ReadAll(in)
 }
+
+var panicHeadlineRegex = regexp.MustCompile(`(panic:|fatal error:)`)
+
+func kubeletPanicDetected(nodeName, logLine string) monitorapi.Intervals {
+	if !panicHeadlineRegex.MatchString(logLine) {
+		return nil
+	}
+
+	failureTime := utility.SystemdJournalLogTime(logLine, time.Now().Year())
+	nodeLocator := monitorapi.NewLocator().NodeFromName(nodeName)
+
+	return monitorapi.Intervals{
+		monitorapi.NewInterval(monitorapi.SourceKubeletLog, monitorapi.Error).
+			Locator(nodeLocator).
+			Message(monitorapi.NewMessage().Reason(monitorapi.KubeletPanic).
+				HumanMessage("kubelet panic detected, check logs for details")).
+			Display().
+			Build(failureTime, failureTime.Add(1*time.Second)),
+	}
+}
+
+// eventsFromCrioLogs returns the produced intervals from CRI-O logs.
+// Right now it only detects panics, but more detectors can be added as needed.
+func eventsFromCrioLogs(nodeName string, crioLog []byte) monitorapi.Intervals {
+	ret := monitorapi.Intervals{}
+
+	scanner := bufio.NewScanner(bytes.NewBuffer(crioLog))
+	for scanner.Scan() {
+		currLine := scanner.Text()
+		ret = append(ret, crioPanicDetected(nodeName, currLine)...)
+	}
+
+	return ret
+}
+
+func crioPanicDetected(nodeName, logLine string) monitorapi.Intervals {
+	if !panicHeadlineRegex.MatchString(logLine) {
+		return nil
+	}
+
+	failureTime := utility.SystemdJournalLogTime(logLine, time.Now().Year())
+	nodeLocator := monitorapi.NewLocator().NodeFromName(nodeName)
+
+	return monitorapi.Intervals{
+		monitorapi.NewInterval(monitorapi.SourceCrioLog, monitorapi.Error).
+			Locator(nodeLocator).
+			Message(monitorapi.NewMessage().Reason(monitorapi.CrioPanic).
+				HumanMessage("CRI-O panic detected, check logs for details")).
+			Display().
+			Build(failureTime, failureTime.Add(1*time.Second)),
+	}
+}
```

Comment:
is there any reason we can't factor our kubelet and cri-o panic detection into one function? they look duplicated